designer/client/src/components/toolbars/search/utils.ts

```diff
@@ -92,18 +113,100 @@ export function useFilteredNodes(filterValues: string[]): {
+                                .concat(edges.length ? "edgeExpression" : null)
+                                .map((name) => displayNames[name])
+                                .filter(Boolean);
+
+                            console.log(groups);
+
+                            return { node, edges, groups };
+                        }
+                        case SearchType.ADVANCED: {
+                            edges =
+                                "edgeExpression" in searchOption
+                                    ? allEdges
+                                          .filter((e) => e.from === node.id)
+                                          .filter((e) => matchFilters(e.edgeType?.condition, searchOption.edgeExpression))
+                                    : [];
+
+                            const keyNamesRelevantForFiltering = Object.keys(searchOption).filter((key) => key !== "searchType");
+
+                            const displayKeyNamesRelevantForFiltering: string[] = keyNamesRelevantForFiltering.map(
+                                (name) => displayNames[name],
+                            );
 
-                    const groups = findFields(filterValues, node)
-                        .concat(edges.length ? "edgeExpression" : null)
-                        .map((name) => displayNames[name])
-                        .filter(Boolean);
+                            groups = keyNamesRelevantForFiltering
+                                .map((key) => findFieldsUsingSelectorWithName(key, searchOption[key], node))
+                                .flat()
+                                .concat(edges.length ? "edgeExpression" : null)
+                                .map((name) => displayNames[name])
+                                .filter(Boolean);
 
-                    return { node, edges, groups };
+                            const allChecksPassed = arraysEqual(groups, displayKeyNamesRelevantForFiltering);
+
+                            if (!allChecksPassed) {
+                                edges = [];
+                                groups = [];
+                            }
+
+                            return { node, edges, groups };
+                        }
+                    }
                 })
                 .filter(({ groups }) => groups.length),
-        [displayNames, allEdges, filterValues, allNodes],
+        [displayNames, allEdges, searchOption, allNodes],
     );
 }
+
+export function resolveSearchOption(filterRawText: string): SearchOption {
+    const advancedFilterKeyNames: string[] = [
+        "id:(",
+        "description:(",
+        "type:(",
+        "paramName:(",
+        "paramValue:(",
+        "outputValue:(",
+        "edgeExpression:(",
+    ]; //heuristic, but sufficient
+    const isAdvancedSearchOption = advancedFilterKeyNames.some((advancedKeyName) => filterRawText.includes(advancedKeyName));
+
+    if (isAdvancedSearchOption) {
```

Comment:
It should work a little differently, basic search shouldn't be cleared but advanced search should be included and narrow the final result the same way as Gmail 

https://github.com/user-attachments/assets/f1e8f275-0e45-4d5c-9204-d2b0a2387b47